src/mqttgw.erl

```diff
@@ -858,6 +858,7 @@ validate_envelope(Val) ->
 
     true = is_binary(Payload),
     true = is_map(Properties),
+    true = lists:all(fun({_, V}) -> is_binary(V) end, maps:to_list(Properties)),
```

Comment:
Only **user properties** are utf8 strings. Please, open the issue and I'll take a closer look when come back.

Reply:
It seems the `status` should be in payload.

Reply:
You're right. We have to validate all the properties according to https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901029

Reply:
https://github.com/netology-group/mqtt-gateway/issues/38